spynnaker_integration_tests/test_local_only/test_network_fake_retina_input.py

```diff
@@ -0,0 +1,222 @@
+                if last_target_time != target_time:
+                    raise Exception(
+                        f"Spike in source {s_label}: {x}, {y} at time {time}"
+                        " does not have matching time in all surrounding"
+                        " targets")
+            found_spikes.append((x_t, y_t, target_time))
+            # print(f"Spike in source {s_label}: {x}, {y} time {time} matches"
+            #       f" target {t_label}: {x_t}, {y_t} time {target_time}")
+    return spikes, found_spikes
+
+
+def do_run():
+    # Set up PyNN
+    p.setup(1.0)
+
+    # Set the number of neurons per core to a rectangle
+    # (creates 512 neurons per core)
+    p.set_number_of_neurons_per_core(p.IF_curr_exp, (SUB_WIDTH, SUB_HEIGHT))
+    # (creates 2048 sources per core)
+    p.set_number_of_neurons_per_core(p.SpikeSourceArray,
+                                     ((SUB_WIDTH * 2), (SUB_HEIGHT * 2)))
+
+    # This is our convolution connector.  This one doesn't do much!
+    conn = p.ConvolutionConnector([[WEIGHT, WEIGHT, WEIGHT],
+                                   [WEIGHT, WEIGHT, WEIGHT],
+                                   [WEIGHT, WEIGHT, WEIGHT]], padding=(1, 1))
+
+    capture_conn = p.ConvolutionConnector([[WEIGHT]])
+
+    # This is a fake retina
+    dev = p.Population(
+        WIDTH * HEIGHT, p.SpikeSourceArray(get_retina_input()),
+        structure=Grid2D(WIDTH / HEIGHT), label="Spike Source")
+
+    # Create a population that captures the spikes from the input
+    capture = p.Population(
+        WIDTH * HEIGHT, p.IF_curr_exp(), structure=Grid2D(WIDTH / HEIGHT),
+        label="Capture")
+    p.Projection(dev, capture, capture_conn, p.Convolution())
+
+    # Create some convolutional "layers" (just 2, with 1 convolution each here)
+    pop = p.Population(
+        WIDTH * HEIGHT, p.IF_curr_exp(), structure=Grid2D(WIDTH / HEIGHT),
+        label="Layer One")
+    pop_2 = p.Population(
+        WIDTH * HEIGHT, p.IF_curr_exp(), structure=Grid2D(WIDTH / HEIGHT),
+        label="Layer Two")
+
+    # Record the spikes so we know what happened
+    capture.record("spikes")
+    pop.record("spikes")
+    pop_2.record("spikes")
+
+    # Create convolution connections from the device -> first pop -> second pop
+    # These use the same connector, but could be different if desired
+    p.Projection(dev, pop, conn, p.Convolution())
+    p.Projection(pop, pop_2, conn, p.Convolution())
+
+    # Run the simulation for long enough for packets to be sent
+    p.run(run_time)
+
+    # Get out the spikes
+    capture_spikes = capture.get_data("spikes").segments[0].spiketrains
+    layer_1_spikes = pop.get_data("spikes").segments[0].spiketrains
+    layer_2_spikes = pop_2.get_data("spikes").segments[0].spiketrains
+
+    # Tell the software we are done with the board
+    p.end()
+
+    return (capture_spikes, layer_1_spikes, layer_2_spikes)
+
+
+class SingleSpikeKernelResponse(BaseTestCase):
```

Comment:
Just noticed this class has the wrong name, it should be NetworkFakeRetinaInput or something like that.